### PR TITLE
[WIP] Unified and converted the cloud network form into a function component

### PR DESCRIPTION
**Pull request by skateman**
I joined the `create` and `update` forms into a single one, determining the action based on the `recordId` prop. Also converted the old class-based component into a function-component using hooks. I'm not sure about the loading part, but there's a lot of repetition among other forms and I think we could come up with a wrapper that would abstract away all the common pieces. cc @himdel

I still don't have credentials to the NSX-T provider, so this is staying WIP :confused:

### CODE REVIEW

**Comment by chessbyte** [comment]
@skateman would you like @gubbe505 to test this out in a real NSX-T environment?

**Comment by gubbe505** [comment]
I wil check the changes with a live NSX-T environment.

**Review by gubbe505 — COMMENTED** [review]
In the CloudNetworkCenter and CloudNetworkCenters the component_name have to be changed to NsxtCloudNetworkForm.

The field subnet_mask and security_zone are only in create mode

File: app/javascript/components/cloud-networks/nsxt-cloud-network-form.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    ManageIQ.redux.store.dispatch({
+      type: 'FormButtons.callbacks',
+      payload: { addClicked: () => formOptions.submit() },
+    });
+  };
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+    const request = recordId ? API.patch(`/api/cloud_networks/${recordId}`, values) : API.post('/api/cloud_networks', values);
```
[the remark this answers is not in the document]
Comment (skateman) [reply app/javascript/components/cloud-networks/nsxt-cloud-network-form.jsx:42]: https://github.com/ManageIQ/manageiq-api/pull/927

**Pull request closed without merging**